### ISSUE TITLE
Support cross-compilation

### ISSUE THREAD

**Issue by roblabla** [volks73/cargo-wix#110]
## Motivation

Currently, it appears to be impossible to do something like `cargo wix --target=i686-pc-windows-msvc` in order to build 32-bit target platforms (or, to take it a step further, using aarch64 or arm targets to make ARM installers). The platform is hardcoded [here](https://github.com/volks73/cargo-wix/blob/master/src/create.rs#L791). This means to create a 32-bit installer, we must first install a 32-bit build of cargo-wix! It's like the gcc canadian cross silliness all over again! :scream:

## Proposed solution

I propose that we add a new flag to cargo create, `--target=targetname`. This flag will be passed through to cargo when building. Cargo will then produce its output to the `target/$(targetname)/release/` folder. The target will then be passed through to `wix` using the `var.CargoTarget` variable. If target is unset, `var.CargoTarget` will be an empty string.

When CargoTarget is set, [platform](https://github.com/volks73/cargo-wix/blob/master/src/create.rs#L791) will parse it to figure out which platform we're building for automatically (by looking at the first component of the target platform). (NOTE: Platform and CargoTarget aren't redundant. CargoTarget will be *unset* when doing a "host build", but Platform will still be set to x86 or x64 depending on the host platform, allowing wix to know if this is a 64-bit build).

Finally, the templates will be changed such that binary paths will be `$(var.CargoTargetDir)\$(var.CargoTarget)\$(var.Profile)\binaryname`.

When the CargoTarget is set, this will expand to `target\x86_64-pc-windows-msvc\release\binaryname`. If it is unset (e.g. doing a host build), it will expand to `target\\release\binaryname`, which should still work properly (:crossed_fingers:).

This way, making a 32-bit and 64-bit installers will be as simple as running

```
cargo wix --target x86_64-pc-windows-msvc
cargo wix --target i686-pc-windows-msvc
```

**Comment by volks73** [comment]
The `--target,-t` option is also not taken, so now change to the current CLI will be needed, just addition.

This all seems reasonable. Looking at the hard-coded platform implementation, it appears kind of unfinished. My guess is that I was supposed to circle back to code and make it more flexible and to account for different architectures, but I wanted something to work first.

**Comment by volks73** [comment]
Thank you for the contribution and discussion! Closed as of 703a0f1186fb762b8cd7227e6d306a3db24a45ef via merge of #113.
